examples/twittermap/web/public/javascripts/sidebar/controllers.js

```diff
@@ -79,31 +94,124 @@ angular.module("cloudberry.sidebar", ["cloudberry.common"])
+
+          if($scope.drawTweetMode==1){
+            sampleTweets = [];
+            sampleTweets = resultSet[0];
+            drawTweetsTraditional();
+            setTimeout(function(){
+                $("#tweet").children().filter("twitterwidget").removeClass("twitter-tweet").css("opacity","0").animate({opacity:1},1000);
+            },1000);
+          }
+          else{
+            sampleTweets = sampleTweets.concat(resultSet[0]);//oldest tweet will be at front
+          }
       }, "sampleTweetsRequest");
-      $scope.isSampleTweetsOutdated = false;
+      
+      
+    }
+  
+    function drawTweets(message) {           
+      var url = "https://api.twitter.com/1/statuses/oembed.json?callback=JSON_CALLBACK&id=" + message["id"];
+      $http.jsonp(url).success(function (data) { 
+        $(data.html).hide().prependTo("#tweet");
+        $("#tweet").children().filter("twitterwidget").first().removeClass("twitter-tweet").hide().slideDown(1000);
+      });
+      
     }
+    function drawTweetsTraditional(){
+      $.each(sampleTweets, function (i, d) {
+      var url = "https://api.twitter.com/1/statuses/oembed.json?callback=JSON_CALLBACK&id=" + d.id;
+      $http.jsonp(url).success(function (data) {
+          $(data.html).hide().prependTo("#tweet");
+        });
+      });
+    }  
+    //Constantly checking local tweets queue to draw tweet one by one
+    function startLiveTweet(){        
+      $scope.liveTweetsLoop = window.setInterval(function(){
+        if(sampleTweets.length>0){
+          var data = sampleTweets.pop();
+          drawTweets(data);
+        }
+        if($("#tweet").children().length>20)
+        {
+          $("#tweet").children().last().remove();
+        }
+      },3000);
+    };
+  
+    function cleanLiveTweet(){
+      window.clearInterval($scope.liveTweetsLoop);
+      $("#tweet").html("");//clean tweets in sidebar
+      sampleTweets=[];
+    };
+
+    function handleSidebarQuery(){  
 
-    function handleSidebarQuery() {
+      var timeBarMin = new Date(cloudberry.parameters.timeInterval.start);//user specified time series start
+      var timeBarMax = new Date(cloudberry.parameters.timeInterval.end);//user specified time series end
+      //Clear both query and updating loop of live Tweets
 
+      if(secondLiveTweetQueryTimeOut){
+        clearTimeout(secondLiveTweetQueryTimeOut);
+      }
+
+      
       if ($scope.isHashTagOpen && $scope.isHashTagOutdated) {
         sendHashTagQuery();
       }
-
+      
       if ($scope.isSampleTweetsOpen && $scope.isSampleTweetsOutdated) {
-        sendSampleTweetsQuery();
+        
+        cleanLiveTweet();
+        //Do traditional sample tweets,when user specifed time interval, and the end of time interval is older than latest tweet
+        if(timeBarMax<timeSeriesEnd){
+          
+          $scope.drawTweetMode = 1;
+          sendSampleTweetsQuery(timeBarMin.toISOString(),timeBarMax.toISOString(),10,mode=1);
```

Comment:
```suggestion
          sendSampleTweetsQuery(timeBarMin.toISOString(), timeBarMax.toISOString(), 10);
```

Reply:
There's no `mode` parameter in the definition of `sendSampleTweetsQuery()`at all.